api/openapi-spec.yaml

```diff
@@ -0,0 +1,473 @@
+
+          **Examples nested structures:**
+          ```
+          /events/?data.identity=pkg:maven/my.namespace/my-name@1.0.0
+          /events/?meta.source.domainId=my.domain&data.identity=pkg:maven/my.namespace/my-name@1.0.0    #Multiple keys and nested structures
+          ```
+
+          Note that mutiple keys only is allowed with logical AND (via `&`). There are no support for logical OR.
+
+          Filters also allow for regex to be used. This means you can make more complex matchings or use partial matching. In regex `.*` matches anything and basically works like wildcards.
+
+          **Examples of regex:**
+          ```
+          /events/?data.identity=my-artifact@1.0.0              #Partial match using artifact name and version
+          /events/?data.identity=my.namespace                   #Partial match using artifact namespace
+          /events/?data.identity=my.namespace/.*@1.0.0          #Matches any version 1.0.0 artifact within the namespace 'my.namespace'
+          /events/?data.identity=my.namespace/my-name@.*        #Matches any version of the artifact 'my-artifact' within the namepsace 'my-namespace'
+          /events/?data.identity=my.namespace/.*                #Matches any artifact within the namespace 'my.namespace'
+          ```
+          **Data types:**
+
+          By default, all data types are treated as strings. It is possible to specify the data type by including it explicitly in the query:
+
+          ```
+          /events/?int(key1)>=value1&key2!=value2&double(key3)<value3
+          ```
+
+          In this example, `key1` and `value1` are treated as integers, `key2` and `value2` are treated as strings and `key3` and `values3` are treated as doubles.
+
+          Currently only string (default), integer and double data types are supported.
+
+          **No comparator:**
+
+          Note that for events query, there is one special supported case is that the query expression has no comparator and value at all, for example `/events/?key`, which means getting all events who has `key` as a field in the JSON documents. The REST API for events query also supports `!key` as a filter which means getting all events without field ‘key’ in the JSON documents. Here are some examples:
+
+          ```
+          /events/?data.identity                                   #Get all the events containing field 'identity' in the 'data'
+          /events/?meta.source.domainId=my.domain&data.identity    #DomainId is 'my.domain' and has field 'identity' in 'data'
+          /events/?!data.customData                                #Return all the events without field 'customData' in 'data'
+          ```
+
+        schema:
+          type: object
+          additionalProperties:
+            type: string
+        style: form
+        explode: true
+        example: |
+          {
+            "meta.type":"EiffelArtifactCreatedEvent",
+            "data.identity":"pkg:maven/my.namespace/my-name@1.0.0"
+          }
```

Comment:
If you try the swagger editor it might be a bit more clear. This is the syntax for specifying a list of parameters to send in the request. The above object will add `&meta.type=EiffelArtifactCreatedEvent&data.identity=pkg%3Amaven%2Fmy.namespace%2Fmy-name%401.0.0`to the URL.

Don't know if we should add some comment here to make it more visible.

Reply:
Oh. Yeah, that certainly wasn't obvious to me. If we could make that a bit more clear that'd probably be a good idea.

Reply:
Tried to explain a bit in a9debeb

Reply:
Tried to explain in a9debeb